_posts/2019-02-04-js-calculator.html

```diff
@@ -0,0 +1,59 @@
+---
+title: Vanilla js calculator
+thumbnail: -SITE_URL-/images/calculator.jpg
+excerpt: "A copy of the windows calculator functionality. It is meant to be an exercise in vanilla javascript."
+# githublink: "#"
+bitbucketlink: "https://bitbucket.org/Dries_Sercu/vanilla-js-calculator"
+demolink: "-SITE_URL-/projects/vanilla-js-calculator/index.html"
+---
+<p>
+	It had been a while since I made anything using javascript. As an exercise I reproduced as much of the windows calculator functionality as possible.
+</p>
+<p>
+	I decided against using a framework because I wanted to focus on javascript instead of some framework. Using a framework might have made this easier but the focus of the exercise would have been different.
+</p>
+<p>
+	While I was at it I also decided to write tests and to generate reports about my tests.
+</p>
+<p>
+	The calculator has some nice features, and it's been a nice study in user experience.
+</p>
+
+<h2>Jasmine</h2>
+
+<p>
+	From previous experience i learned that, without realizing it, you will break your own application.
+	To make sure I could break pieces without knowing about it something was required to verify that every piece of code worked as expected.
+	Furthermore, the faster I knew something broke, the better. Using any testing framework prevents you from having to go to your browser and clicking every button every time you make a change.
+	With this in mind I implemented <a href="https://jasmine.github.io/">Jasmine</a>.
+</p>
+<p class="quote">
+	"Jasmine is a behavior-driven development framework for testing JavaScript code."
+</p>
+<p>
+	In practice, this means that anytime I modified my code Jasmine would tell me whether or not my code still worked. There's a catch though, I have to tell Jasmine what to test. This means that if I forget something or I make a mistake in one of my tests, this will be reflected in the report.
```

Comment:
In practice, this means that anytime I modified my code Jasmine would tell me whether or not my code still worked. There's a catch though: I have to tell Jasmine what to test. This means that if I forget something or I make a mistake in one of my tests, this will be reflected in the report.